### PR TITLE
Update theme builder dependencies and fix warnings/lint

### PR DESCRIPTION
**Pull request by ricknout**
Also makes the "Theme name..." `TextView` in `activity_main` an `EditText`, not sure why it wasn't one before...

### CODE REVIEW

File: MaterialThemeBuilder/app/build.gradle [hunk 1]
```diff
@@ -30,15 +30,14 @@ dependencies {
     implementation fileTree(dir: 'libs', include: ['*.jar'])
 
     // Kotlin
-    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
+    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8:$kotlin_version"
```
Comment (chrisbanes) [inline MaterialThemeBuilder/app/build.gradle:33]: You probably don't need the `jdk8` dep, does `kotlin-stdlib` work?

File: MaterialThemeBuilder/app/src/main/java/io/material/materialthemebuilder/ui/MainActivity.kt [hunk 2]
```diff
@@ -47,7 +46,7 @@ class MainActivity : AppCompatActivity() {
     viewPager.adapter = adapter
 
     (application as App).preferenceRepository
-      .nightModeLive.observe(this, Observer { nightMode ->
+      .nightModeLive.observe(this, { nightMode ->
```
Comment (chrisbanes) [inline MaterialThemeBuilder/app/src/main/java/io/material/materialthemebuilder/ui/MainActivity.kt:49]: Move the lambda outside of the parens?

File: MaterialThemeBuilder/app/src/main/java/io/material/materialthemebuilder/ui/instruction/InstructionsFragment.kt [hunk 3]
```diff
@@ -44,7 +43,7 @@ class InstructionsFragment : Fragment() {
     val darkThemeSwitch: SwitchMaterial = view.findViewById(R.id.dark_theme_switch)
     val preferenceRepository = (requireActivity().application as App).preferenceRepository
 
-    preferenceRepository.isDarkThemeLive.observe(this, Observer { isDarkTheme ->
+    preferenceRepository.isDarkThemeLive.observe(this, { isDarkTheme ->
```
Comment (chrisbanes) [inline MaterialThemeBuilder/app/src/main/java/io/material/materialthemebuilder/ui/instruction/InstructionsFragment.kt:46]: Ditto. Move lambda out

**Review by chrisbanes — COMMENTED** [review]
LGTM, just a few nits

**Review by chrisbanes — APPROVED** [review]

**Review by hunterstich — APPROVED** [review]
Amazing! Thanks Nick

**Pull request merged**